app/services/search_rummager_service.rb

```diff
@@ -1,17 +1,21 @@
 class SearchRummagerService
   def fetch_related_documents(filter_params = {})
-    options = default_search_options.merge(filter_params)
+    search_params = default_search_params.merge(filter_params)
 
-    Rails.cache.fetch(options, expires_in: 5.minutes) do
-      search_response = Whitehall.search_client.search(options)
-      search_response["results"].map! { |res| RummagerDocumentPresenter.new(res) }
-      search_response
-    end
+    return search_response(search_params) if Rails.env.test?
```

Comment:
Do you know why this is useful? Ideally the code paths wouldn't differ based on the Rails environment.

Reply:
Yeah I am not a fan of this either.

This is an attempt to get around: `LatestControllerTest#test_GET_:index_should_accept_pagination_parameters:
TypeError: no _dump_data is defined for class Thread::Mutex` (https://ci.integration.publishing.service.gov.uk/job/whitehall/job/upgrade-rails-to-version-5.2.3/68/console)

The alternative would be to use dependency injection to use an object in our test that does not use caching.

Reply:
I would have expected `config.action_controller.perform_caching = false` in  `config/environments/test.rb` to prevent this, but no.

Reply:
@cbaines  `SearchRummagerService`' is a thin wrapper around the search client, whose main function seems to be: 

* cache results returned bu the search client
* merge default search options into the the provided filter params

So isn't [this test](https://github.com/alphagov/whitehall/blob/master/test/unit/services/search_rummager_service_test.rb#L13) testing the search client itself rather than the object under test?

Reply:
I can see this has been addressed in https://github.com/alphagov/whitehall/pull/4998/commits/4314f729feb1b357faca93878009cb306cdb2056 now, thanks :)